enterprise/backend/src/metabase_enterprise/query_field_validation/api.clj

```diff
@@ -0,0 +1,99 @@
+(ns metabase-enterprise.query-field-validation.api
+  (:require
+   [compojure.core :refer [GET]]
+   [metabase.api.common :as api]
+   [metabase.api.routes.common :refer [+auth]]
+   [metabase.server.middleware.offset-paging :as mw.offset-paging]
+   [toucan2.core :as t2]))
+
+(def ^:private valid-sort-columns "Columns that the card errors can be sorted by"
+  #{"name" "collection" "created_by" "last_edited_at"}) ;; We don't support sorting by error message for now
+
+(def ^:private default-sort-column "name")
+
+(def ^:private valid-sort-directions
+  #{"asc" "desc"})
+
+(def ^:private default-sort-direction "asc")
+
+(def ^:private card-joins
+  [[(t2/table-name :model/QueryField) :qf] [:and
+                                            [:= :qf.card_id :c.id]
+                                            [:= :qf.explicit_reference true]]
+   [(t2/table-name :model/Field) :f] [:and
+                                      [:= :qf.field_id :f.id]
+                                      [:= :f.active false]]])
+
+(defn- cards-with-inactive-fields
+  [sort-column sort-direction offset limit]
+  (let [additional-join       (condp = sort-column
```

Comment:
Ha, okay. I went back and forth on this during dev since on the one hand it could be multiple joins but in practice it'll only ever be one. But pluralizing is fine